eng/common/scripts/SemVer.ps1

```diff
@@ -147,24 +144,42 @@ class AzureEngSemanticVersion : IComparable {
     return $versionString;
   }
 
-  [void] IncrementAndSetToPrerelease() {
-    if ($this.IsPrerelease -eq $false)
+  [void] IncrementAndSetToPrerelease($Segment) {
+    if ($this.BuildNumber)
     {
-      $this.PrereleaseLabel = $this.DefaultPrereleaseLabel
-      $this.PrereleaseNumber = 1
-      $this.Minor++
-      $this.Patch = 0
-      $this.IsPrerelease = $true
+      throw "Cannot increment releases tagged with azure pipelines build numbers"
+    }
+
+    if ($this.PrereleaseLabel)
+    {
+      $this.PrereleaseNumber++
     }
     else
     {
-      if ($this.BuildNumber) {
-        throw "Cannot increment releases tagged with azure pipelines build numbers"
+      $this.$Segment++
+      if($Segment -eq "Major") {
+        $this.Minor = 0
+        $this.Patch = 0
       }
-      $this.PrereleaseNumber++
+      if($Segment -eq "Minor") {
+        $this.Patch = 0
+      }
+
+      # If the major version is 0, we don't need a prerelease label
+      if ($this.Major -ne 0)
+      {
+        $this.PrereleaseLabel = $this.DefaultPrereleaseLabel
```

Comment:
I wonder if this should be also an option as folks like JS that increment the patch don't want to turn it into a prerelease as part of the increment.

Reply:
Maybe a separate function then:

```powershell
[void] IncrementAndSetToPrerelease($Segment) {}
[void] Increment($Segment) {}
```